[Bug]: Internal Server Error When Deploying App

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

A cloud user gets an internal server error when trying to click the Deploy button for an app. Even if they try to fork it, they are not able to deploy the forked app.

Status 500:
`{"responseMeta":{"status":500,"success":false,"error":{"code":5000,"message":"Internal server error while processing request"}}}`

Furthermore the following errors can be seen in the console:
```
POST https://app.appsmith.com/api/v1/applications/publish/.....................................? 500 (Internal Server Error)
main.87faf633.js:2 Error: Internal server error while processing request
at main.87faf633.js:2:1040527
at d (main.87faf633.js:2:9046090)
at Generator._invoke (main.87faf633.js:2:9045843)
at [Generator.next](http://generator.next/) (main.87faf633.js:2:9046453)
at d (main.87faf633.js:2:1274405)
at c (main.87faf633.js:2:1274740)
at main.87faf633.js:2:1269349
at Mr (main.87faf633.js:2:1266908)
at zr (main.87faf633.js:2:1267114)
at Dr (main.87faf633.js:2:1266962)
at Jr.<computed> (main.87faf633.js:2:1269245)
at main.87faf633.js:2:1273998
at p (main.87faf633.js:2:1274959)
at d (main.87faf633.js:2:1274465)
at di (main.87faf633.js:2:1274198)
at main.87faf633.js:2:1273936
at p (main.87faf633.js:2:1274959)
at d (main.87faf633.js:2:1274465)
at c (main.87faf633.js:2:1274740)
```

### Steps To Reproduce

https://app.frontapp.com/open/cnv_wcyqpu4?key=Fzrrn5ZtStYcdXYJuFs8XGLn3t26klSU

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Cloud

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_7i0gc)

--- a/app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Layout.java
+++ b/app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Layout.java
@@ -58,9 +58,6 @@ public class Layout extends BaseDomain {
     Set<String> allOnPageLoadActionNames;
 
     @JsonIgnore
-    Set<ActionDependencyEdge> allOnPageLoadActionEdges;
-
-    @JsonIgnore
     Set<String> actionsUsedInDynamicBindings;
 
     @JsonIgnore
@@ -90,7 +87,6 @@ public class Layout extends BaseDomain {
         this.setAllOnPageLoadActionNames(null);
         this.setCreatedAt(null);
         this.setUpdatedAt(null);
-        this.setAllOnPageLoadActionEdges(null);
         this.setActionsUsedInDynamicBindings(null);
         this.setWidgetNames(null);
         List<Set<DslActionDTO>> layoutOnLoadActions = this.getLayoutOnLoadActions();

--- a/app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration004ResetOnPageLoadEdgesInLayouts.java
+++ b/app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration004ResetOnPageLoadEdgesInLayouts.java
@@ -9,7 +9,15 @@ import org.springframework.data.mongodb.core.query.Query;
 import org.springframework.data.mongodb.core.query.Update;
 import org.springframework.data.mongodb.core.query.UpdateDefinition;
 
-
+/**
+ * Since an Appsmith application is not limited by the number of interconnections it can make,
+ * it is possible for users to create graphs so complex that we end up having MBs of data in a
+ * relatively simple page as well.
+ * This was causing failures in a user's app since they were now unable to update their app
+ * any further. We do not really use this property anywhere in our logic today,
+ * and this is essentially redundant information.
+ * This migration gets rid of the edges property from the layout in all existing pages.
+ */
 @ChangeUnit(order = "004", id = "reset-on-page-load-edges-in-layouts")
 public class Migration004ResetOnPageLoadEdgesInLayouts {
     private final MongoTemplate mongoTemplate;

--- a/app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration004ResetOnPageLoadEdgesInLayouts.java
+++ b/app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration004ResetOnPageLoadEdgesInLayouts.java
@@ -1,0 +1,33 @@
+package com.appsmith.server.migrations.db.ce;
+
+import com.appsmith.server.domains.NewPage;
+import io.mongock.api.annotations.ChangeUnit;
+import io.mongock.api.annotations.Execution;
+import io.mongock.api.annotations.RollbackExecution;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.data.mongodb.core.query.UpdateDefinition;
+
+
+@ChangeUnit(order = "004", id = "reset-on-page-load-edges-in-layouts")
+public class Migration004ResetOnPageLoadEdgesInLayouts {
+    private final MongoTemplate mongoTemplate;
+
+    public Migration004ResetOnPageLoadEdgesInLayouts(MongoTemplate mongoTemplate) {
+        this.mongoTemplate = mongoTemplate;
+    }
+
+    @RollbackExecution
+    public void demoRollbackExecution() {
+    }
+
+    @Execution
+    public void executeMigration() {
+        UpdateDefinition updateQuery = new Update()
+                .unset("unpublishedPage.layouts.0.allOnPageLoadActionEdges")
+                .unset("publishedPage.layouts.0.allOnPageLoadActionEdges");
+
+        mongoTemplate.updateMulti(new Query(), updateQuery, NewPage.class);
+    }
+}

--- a/app/server/appsmith-server/src/main/java/com/appsmith/server/models/export/LayoutMetadata.java
+++ b/app/server/appsmith-server/src/main/java/com/appsmith/server/models/export/LayoutMetadata.java
@@ -1,19 +1,16 @@
 package com.appsmith.server.models.export;
 
-import java.util.List;
-import java.util.Set;
-
-import org.json.JSONObject;
-
 import com.appsmith.external.exceptions.ErrorDTO;
-import com.appsmith.server.domains.ActionDependencyEdge;
 import com.appsmith.server.domains.ScreenType;
 import com.appsmith.server.dtos.DslActionDTO;
 import com.fasterxml.jackson.annotation.JsonIgnore;
+import lombok.Data;
+import org.json.JSONObject;
+
+import java.util.List;
+import java.util.Set;
 
 import static java.lang.Boolean.TRUE;
-
-import lombok.Data;
 
 @Data
 public class LayoutMetadata {
@@ -28,7 +25,6 @@ public class LayoutMetadata {
     private List<Set<DslActionDTO>> publishedLayoutOnLoadActions;
     private Set<String> widgetNames;
     private Set<String> allOnPageLoadActionNames;
-    private Set<ActionDependencyEdge> allOnPageLoadActionEdges;
     private Set<String> actionsUsedInDynamicBindings;
     private Set<String> mongoEscapedWidgetNames;
     private Boolean validOnPageLoadActions = TRUE;

--- a/app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/LayoutActionServiceCEImpl.java
+++ b/app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/LayoutActionServiceCEImpl.java
@@ -707,7 +707,6 @@ public class LayoutActionServiceCEImpl implements LayoutActionServiceCE {
                             // in the layout for re-use to avoid computing DAG unnecessarily.
                             layout.setLayoutOnLoadActions(onLoadActions);
                             layout.setAllOnPageLoadActionNames(actionNames);
-                            layout.setAllOnPageLoadActionEdges(edges);
                             layout.setActionsUsedInDynamicBindings(actionsUsedInDSL);
                             // The below field is to ensure that we record if the page load actions computation was valid
                             // when last stored in the database.

--- a/app/server/appsmith-server/src/test/resources/test_assets/ActionCollectionServiceTest/mockObjects.json
+++ b/app/server/appsmith-server/src/test/resources/test_assets/ActionCollectionServiceTest/mockObjects.json
@@ -20,7 +20,6 @@
             "MainContainer"
           ],
           "allOnPageLoadActionNames": [],
-          "allOnPageLoadActionEdges": [],
           "actionsUsedInDynamicBindings": [],
           "deleted": false,
           "policies": []

